src/Controls/src/Core/Handlers/Items/Android/Adapters/StructuredItemsViewAdapter.cs

```diff
@@ -33,12 +33,13 @@ protected override void ItemsViewPropertyChanged(object sender, PropertyChangedE
 		{
 			base.ItemsViewPropertyChanged(sender, property);
 
-			if (property.Is(Microsoft.Maui.Controls.StructuredItemsView.HeaderProperty))
+			if (property.Is(Microsoft.Maui.Controls.StructuredItemsView.HeaderProperty) || property.Is(Microsoft.Maui.Controls.StructuredItemsView.HeaderTemplateProperty))
```

Comment:
The test `HeaderFooterTemplateWorks` is failing on Android, could you verify if is related with the changes?